interfaces/builtin/history.go

```diff
@@ -0,0 +1,236 @@
+# Telepathy
+########################
+dbus (receive, send)
+	bus=session
+	path=/org/freedesktop/Telepathy,
+dbus (receive, send)
+	bus=session
+	path=/org/freedesktop/Telepathy/Client,
+dbus (receive, send)
+	bus=session
+	path=/org/freedesktop/Telepathy/Client/HistoryDaemonObserver,
+dbus (receive, send)
+	bus=session
+	path=/org/freedesktop/Telepathy/AccountManager,
+`)
+
+var historyConnectedPlugAppArmor = []byte(`
+# Description: Can access the history-service. This policy group is reserved
+#  for vetted applications only in this version of the policy. A future
+#  version of the policy may move this out of reserved status.
+# Usage: reserved
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+#dbus (send)
+#     bus=session
+#     path=/org/freedesktop/DBus
+#     interface=org.freedesktop.DBus
+#     member={Request,Release}Name
+#     peer=(name=org.freedesktop.DBus),
+#dbus (send)
+#     bus=session
+#     path=/org/freedesktop/*
+#     interface=org.freedesktop.DBus.Properties
+#     peer=(label=unconfined)
+dbus (send)
+     bus=session
+     path=/com/canonical/HistoryService
+     peer=(name=com.canonical.HistoryService,label=unconfined),
+dbus (receive)
+     bus=session
+     path=/com/canonical/HistoryService
+     peer=(label=unconfined),
+dbus (send)
+     bus=session
+     path=/com/canonical/HistoryService/**
+     peer=(name=com.canonical.HistoryService,label=unconfined),
+dbus (receive)
+     bus=session
+     path=/com/canonical/HistoryService/**
+     peer=(label=unconfined),
+`)
+
+var historyPermanentSlotSecComp = []byte(`
+# Description: Allow operating as the history service. Reserved because this
+# gives
+#  privileged access to the system.
+# Usage: reserved
+accept
+accept4
+bind
+connect
+getpeername
+getsockname
+getsockopt
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+setsockopt
+shutdown
+socketpair
+socket
+`)
+
+var historyConnectedPlugSecComp = []byte(`
+# Description: Allow using history service. Reserved because this gives
+#  privileged access to the history service.
+# Usage: reserved
+
+# Can communicate with DBus system service
+connect
+getsockname
+recv
+recvmsg
+send
+sendto
+sendmsg
+socket
+`)
+
+var historyPermanentSlotDBus = []byte(`
+<policy user="root">
+    <allow own="com.canonical.HistoryService"/>
+	<allow own="org.freedesktop.Telepathy.Client.HistoryDaemonObserver"/>
+    <allow send_destination="com.canonical.HistoryService"/>
+	<allow send_destination="org.freedesktop.Telepathy.Client.HistoryDaemonObserver"/>
+    <allow send_interface="org.freedesktop.DBus.ObjectManager"/>
+    <allow send_interface="org.freedesktop.DBus.Properties"/>
+</policy>
+<policy context="default">
+    <deny send_destination="com.canonical.HistoryService"/>
+</policy>
+`)
+
+type HistoryInterface struct{}
+
+func (iface *HistoryInterface) Name() string {
+	return "history"
```

Comment:
We need something more clear and specific as a name for this one. Imagine a snap in the wild including `plugs: [history]`. What is the history service?  History of what?